src/GovUk.Education.ExploreEducationStatistics.Data.Services/SubjectMetaService.cs

```diff
@@ -168,21 +162,110 @@ private async Task<SubjectMetaViewModel> GetSubjectMetaViewModel(ReleaseSubject
+                        {
+                            throw new ArgumentException("When fetching TimePeriods, shouldn't provide time periods in request"); // @MarkFix
+                        }
+
+                        var stopwatch = Stopwatch.StartNew();
+
+                        var observations = _statisticsDbContext
+                            .Observation
+                            .AsNoTracking()
+                            .Where(o => o.SubjectId == query.SubjectId);
+
+                        if (!query.LocationIds.IsNullOrEmpty())
+                        {
+                            observations = observations.Where(o => query.LocationIds.Contains(o.LocationId));
+                        }
+
+                        var timePeriods = GetTimePeriods(observations);
+
+                        _logger.LogTrace("Got Time Periods in {Time} ms", stopwatch.Elapsed.TotalMilliseconds);
+
+                        return new SubjectMetaViewModel
+                        {
+                            TimePeriod = timePeriods,
+                        };
+                    }
+                    case SubjectMetaQueryStep.FilterItems:
+                    {
+                        var stopwatch = Stopwatch.StartNew();
+
+                        var observations =
+                            await _observationService.GetMatchedObservations(query, cancellationToken);
+                        _logger.LogTrace("Got Observations in {Time} ms", stopwatch.Elapsed.TotalMilliseconds);
+                        stopwatch.Restart();
+
+                        var filterItems = await
+                            _filterItemRepository.GetFilterItemsFromMatchedObservationIds(query.SubjectId,
+                                observations);
+                        var filters =
+                            FiltersViewModelBuilder.BuildFiltersFromFilterItems(filterItems,
+                                releaseSubject.FilterSequence);
+                        _logger.LogTrace("Got Filters in {Time} ms", stopwatch.Elapsed.TotalMilliseconds);
+                        stopwatch.Restart();
+
+                        return new SubjectMetaViewModel
+                        {
+                            Filters = filters,
+                        };
+                    }
+                    case SubjectMetaQueryStep.Indicators:
+                    {
+                        var stopwatch = Stopwatch.StartNew();
+
+                        var indicators = await GetIndicators(releaseSubject);
+                        _logger.LogTrace("Got Indicators in {Time} ms", stopwatch.Elapsed.TotalMilliseconds);
+
+                        return new SubjectMetaViewModel
+                        {
+                            Indicators = indicators,
+                        };
+                    }
+                };
+
+            }
+
+            // NOTE: If IncludeInResponse not set, revert to previous behaviour...
```

Comment:
Why are we maintaining the old behaviour? Subject meta filtering only happens in table tool and we control how this endpoint is used so there's no need to maintain this at all.

Reply:
This is still in as a consequence of not doing the frontend work to separate filter and indicators steps in the same PR. So the request made in the final table tool step still needs the old functionality. 

Once the frontend work is done to separate them, then this can be removed. I'll update the comment so this is clearer.